### ISSUE TITLE
Input and output shape of `torch.ops.quantized.batch_norm1d(_relu)?` do not match for 2d inputs 

### ISSUE THREAD

**Issue by pmeier** [pytorch/pytorch#59200]
## 🐛 Bug

Input and output shape of `torch.ops.quantized.batch_norm1d(_relu)?` do not match for 2d inputs 

## To Reproduce

```python
import torch

shapes = (1, 1)
dtype = torch.quint8

X = torch.rand(*shapes, dtype=torch.float) - 0.5
min_val = torch.min(X)
max_val = torch.max(X)
X_zero_point = int(torch.randint(-128, 127, (1,)))
num_bins = 2 ** 8
X_scale = float(max_val - min_val) / num_bins

c = X.shape[1]
mean = torch.rand(c).float()
var = torch.rand(c).float()
weight = torch.rand(c).float()
bias = torch.rand(c).float()
eps = 0.001

Y_zero_point = 1
Y_scale = 0.5

qx = torch.quantize_per_tensor(X, X_scale, X_zero_point, dtype)
qy = torch.ops.quantized.batch_norm1d_relu(qx, weight, bias, mean, var, eps, Y_scale, Y_zero_point)

assert qx.shape == qy.shape
```

## Environment

```
PyTorch version: 1.9.0a0+gitc9de3e3
Is debug build: True
CUDA used to build PyTorch: Could not collect
ROCM used to build PyTorch: N/A

OS: Arch Linux (x86_64)
GCC version: (crosstool-NG 1.24.0.133_b0863d8_dirty) 9.3.0
Clang version: Could not collect
CMake version: version 3.20.2
Libc version: glibc-2.9

Python version: 3.6 (64-bit runtime)
Python platform: Linux-5.12.5-arch1-1-x86_64-with-arch
Is CUDA available: False
CUDA runtime version: 11.1.105
GPU models and configuration: GPU 0: NVIDIA GeForce GTX 1080
Nvidia driver version: 465.31
cuDNN version: Probably one of the following:
/usr/local/cudnn-8.1.1-cuda-11/lib64/libcudnn.so.8.1.1
/usr/local/cudnn-8.1.1-cuda-11/lib64/libcudnn_adv_infer.so.8.1.1
/usr/local/cudnn-8.1.1-cuda-11/lib64/libcudnn_adv_train.so.8.1.1
/usr/local/cudnn-8.1.1-cuda-11/lib64/libcudnn_cnn_infer.so.8.1.1
/usr/local/cudnn-8.1.1-cuda-11/lib64/libcudnn_cnn_train.so.8.1.1
/usr/local/cudnn-8.1.1-cuda-11/lib64/libcudnn_ops_infer.so.8.1.1
/usr/local/cudnn-8.1.1-cuda-11/lib64/libcudnn_ops_train.so.8.1.1
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.19.5
[pip3] pytest-pytorch==0.0.0
[pip3] torch==1.9.0a0+gitc9de3e3
[conda] magma-cuda111             2.5.2                         1    pytorch
[conda] mkl                       2021.2.0           h726a3e6_389    conda-forge
[conda] mkl-include               2021.2.0           h726a3e6_389    conda-forge
[conda] numpy                     1.19.5           py36h2aa4a07_1    conda-forge
[conda] pytest-pytorch            0.2.0              pyh44b312d_0    conda-forge
[conda] torch                     1.9.0a0+gitc9de3e3           dev_0    <develop>
```

## Additional context

This is tested in 

https://github.com/pytorch/pytorch/blob/6d45d7a6c331ddb856ac34a76bcd3613aa05185b/test/quantization/core/test_quantized_op.py#L2104

and

https://github.com/pytorch/pytorch/blob/6d45d7a6c331ddb856ac34a76bcd3613aa05185b/test/quantization/core/test_quantized_op.py#L2155

but the tests never failed due to a bug in the underlying comparison mechanism. #59067 will disable the failing test case, but it should be reinstated as soon as this issue is fixed.

cc @jerryzh168 @jianyuh @raghuramank100 @jamesr66a @vkuzo @mruberry @VitalyFedyunin @walterddr

**Comment by jerryzh168** [comment]
thanks for reporting, will fix this week

**Comment by pmeier** [comment]
Fixed in  #59264.
